### PR TITLE
feat(build): Add typescript to the content server

### PR DESCRIPTION
**Pull request by shane-tomlinson**
This converts verification-reasons and verification-methods
to typescript. These modules are super simple and can be
converted to enums, a basic data type.

This PR is to prove that adding typescript to the build
works.

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Changing this to ready for review! 

@mozilla/fxa-devs - r?

In particular, @lmorchard, @bbangert, and @ianb might have some feedback from their recent work with Typescript.

File: packages/fxa-content-server/webpack.config.js [hunk 1]
```diff
@@ -45,11 +45,12 @@ const webpackConfig = {
   },
 
   resolve: {
-    extensions: ['.js'],
+    extensions: ['.ts', '.js'],
     modules: [
       path.resolve(__dirname, 'app/scripts'),
       path.resolve(__dirname, 'app/scripts/templates'),
       path.resolve(__dirname, 'app'),
+      path.resolve(__dirname, '.tscompiles/scripts'),
```
Comment (lmorchard) [inline packages/fxa-content-server/webpack.config.js]: Typo here? `s/tscompiles/tscompiled/`

File: packages/fxa-content-server/app/scripts/lib/verification-methods.ts [hunk 2]
```diff
@@ -8,9 +8,11 @@
  * In the future SMS and TOTP could also be included.
  */
 
-export default {
-  EMAIL: 'email',
-  EMAIL_2FA: 'email-2fa',
-  EMAIL_CAPTCHA: 'email-captcha',
-  TOTP_2FA: 'totp-2fa'
+enum VerificationMethods {
```
Comment (bbangert) [inline packages/fxa-content-server/app/scripts/lib/verification-methods.ts]: The Programming Typescript book recommends using const enum's to opt into a stricter form of enums that doesn't allow key lookup that may be unsafe.

File: packages/fxa-content-server/webpack.config.js (hunk 1 above)
Comment (shane-tomlinson) [reply packages/fxa-content-server/webpack.config.js]: Huh, turns out this isn't needed at all!

File: packages/fxa-content-server/app/scripts/lib/verification-methods.ts (hunk 2 above)
Comment (shane-tomlinson) [reply packages/fxa-content-server/app/scripts/lib/verification-methods.ts]: Updated.

Comment (shane-tomlinson) [reply packages/fxa-content-server/app/scripts/lib/verification-methods.ts]: Oh, we have to do key lookup, I reverted that change.

**Comment by bbangert** [comment]
LGTM!

**Pull request merged**